scipy/spatial/transform/rotation.py

```diff
@@ -129,3 +129,35 @@ def as_dcm(self):
             return dcm[0]
         else:
             return dcm
+
+    @classmethod
+    def from_dcm(cls, mat, scalar=True):
+        # For now, mat must be a 3D matrix
+        num_rotations = mat.shape[0]
+        quat = np.empty((num_rotations, 4))
+        if scalar:
+            for rot_num in range(num_rotations):
+                A = mat[rot_num]
+                trA = A.trace()
+                choice = np.argmax(np.array([A[0, 0], A[1, 1], A[2, 2], trA]))
+
+                if choice in [0, 1, 2]:
+                    quat[rot_num, choice] = 1 - trA + 2 * A[choice, choice]
+                    next_ind = (choice + 1) % 3
+                    next_next = (next_ind + 1) % 3
+
+                    quat[rot_num, next_ind] = A[choice, next_ind] + A[next_ind, choice]
+                    quat[rot_num, next_next] = A[choice, next_next] + A[next_next, choice]
+
+                    quat[rot_num, 3] = A[next_ind, next_next] - A[next_next, next_ind]
+                else:
+                    quat[rot_num] = np.array([
+                        A[1, 2] - A[2, 1],
+                        A[2, 0] - A[0, 2],
+                        A[0, 1] - A[1, 0],
+                        1 + trA])
+
+        # For testing purposes only. Return cls(quat) in final implementation.
+        # Creating new objects in python is time consuming. We do not want to
+        # count that time in the benchmarks.
+        return quat
```

Comment:
To be honest I believe there is no problem in creating the object and counting the time. It will be the same for all cases.

Another thing I'm not sure about --- maybe we should normalize `quat` here as an one liner `quat /= np.linalg.norm(quat, axis=1)[:, None]`. Because technically it will avoid copying and norm checks in `__init__`.

Considering this, maybe we should make `__init__` trivial as `self._quat = quat` and delegate `quat` preparation (as normalized 2-d array) wherever else. What do you think about that?

Reply:
Yup, we could pass `normalize=True` to completely skip norm calculations and zero norm checks.

I think `__init__` is fine as is. If we remove preparation from `__init__` there is a possibility that the user may use the `__init__` method to get un-normalized quaternions. While most users will read the documentation and not use `__init__` I'd be more comfortable if we removed that possibility completely. In any case, passing `normalized=True` flag gives us the same functionality and avoids norm calculations  and checks, so I think it's fine as is. Does my reasoning sound sensible? Maybe I'm on the wrong track here.

Reply:
I think you are right, no reason to modify `__init__`. My reasoning was to remove `__init__` as a an initialization method for a user completely, sort of "cleaner" design. But I suggest to keep it. Maybe we will reconsider something later.